ZeeHomeScreen/Classes/Components/SectionComposite/SectionCompositeViewController.swift

```diff
@@ -648,25 +663,27 @@ import Zee5CoreSDK
             let header =  componentModel.componentHeaderModel,
             let layoutName =  header.layoutStyle {
             if let headerView = collectionView.dequeueReusableSupplementaryView(ofKind: kind, withReuseIdentifier: layoutName, for: indexPath) as? UniversalCollectionViewHeaderFooterView {
-                
                 headerView.delegate = self
-                /*if headerView.componentViewController == nil,*/
-                    if let currentModel = header.entry as? APModel {
-                    headerView.componentViewController =  ComponenttFactory.componentViewController(with: header,
-                                                              andModel: currentModel,
-                                                              for: headerView,
-                                                              delegate: self,
-                                                              parentViewController: self)
+
+                if let _ = header.entry as? APModel {
+                    if let componentViewController = ComponenttFactory.viewController(for: header) {
+                        addChild(componentViewController)
```

Comment:
This was handled previously by factory method, I had to change it to comply to what Apple documentation says. Factory method forced the order of execution as:
-addSubview
-addChildViewController
-didMoveToParentViewController

But the correct order according to Apple should be:
-addChildViewController
-addSubview
-didMoveToParentViewController

Because `addChildViewController` already handles the case with calling `will` method, which might do things on a previously attached view, which in the other case was already altered.

Reply:
SRP fix.